src/nix/nix_store.rs

```diff
@@ -84,20 +88,30 @@ impl NixStoreCmd {
     }
 
     /// Return the derivation used to build the given build output.
-    async fn nix_store_query_deriver(&self, out_path: PathBuf) -> Result<DrvOut> {
+    async fn nix_store_query_deriver(&self, out_path: PathBuf) -> Result<DrvOut, NixCmdError> {
         let mut cmd = self.command();
         cmd.args(["--query", "--deriver", &out_path.to_string_lossy().as_ref()]);
         nix_rs::command::trace_cmd(&cmd);
-        let out = cmd.output().await?;
+        let out = cmd
+            .output()
+            .await
+            .map_err(CommandError::ChildProcessError)?;
```

Comment:
I thought this cast automatically happens anyway.

Reply:
Nope. It throws an error saying `? couldn't convert the error to nix-store::NixCmdError`